src/layout.rs

```diff
@@ -71,11 +83,13 @@ impl Event {
 enum State {
     NormalKey { keycode: KeyCode, coord: (u8, u8) },
     LayerModifier { value: usize, coord: (u8, u8) },
+    SequenceKey { keycode: KeyCode },
```

Comment:
The idea is to manage the macro here. for example, we can have a 

```rust
Sequence {
    since: u32, // keep track of timing, incremented at each tick
    delay: u16,
    sequence: &'static [SequenceEvent],
}
```

`since / delay` gives you the index un sequence you're at. Then you can compute the KeyCodes that support this.

if `since / delay > sequence.len()`, tick() would return None, and the state will disapear, and all keys will be automatically released.

Reply:
We don't want all keys to be automatically released though.  We want them to only be released when a specific `Release()` is called.

Reply:
Just to provide an update (and remind myself where I'm at later because I have to step away now for most of the day)...  I've got an implementation using `SequenceKey` inside of `enum State` but it's only working for simple events where a key doesn't need to be held before another key is pressed.   So it can type out a word like, 'foo' but not 'FOO' (because shift would need to be held the whole time).  It seems that something is generating a release event immediately after every press event.  I might have to setup OpenOCD again and whip out gdb because I'm not not finding it.

I'll keep working on it later today.

Reply:
Unit tests are your friends.